code/modules/mob/living/simple_animal/friendly/farm_animals.dm

```diff
@@ -49,14 +49,13 @@
 		LoseTarget()
 		visible_message("<span class='notice'>[src] calms down.</span>")
 
-	if(stat == CONSCIOUS)
-		eat_plants()
-		if(!pulledby)
-			for(var/direction in shuffle(list(1,2,4,8,5,6,9,10)))
-				var/step = get_step(src, direction)
-				if(step)
-					if(locate(/obj/structure/spacevine) in step || locate(/obj/structure/glowshroom) in step)
-						Move(step, get_dir(src, step))
+	eat_plants()

```

Comment:
Won't this mean that goats only eat plants if they're in AI_ON, which, 99% of the time, they're not in? (since its only used in combat).
Won't this make goats useless at eating weeds?

Reply:
It should work now; hacky as it is.